api/metrics/v1/http.go

```diff
@@ -85,7 +85,7 @@ func (n nopInstrumentHandler) NewHandler(labels prometheus.Labels, handler http.
 }
 
 // NewHandler creates the new metrics v1 handler.
-func NewHandler(read, write *url.URL, opts ...HandlerOption) http.Handler {
+func NewHandler(read, write *url.URL, prefixHeader string, opts ...HandlerOption) http.Handler {
```

Comment:
Can we either make this a handler option like the rest? Or pass it directly as a middleware using a `ReadMiddleware` option? This would help us avoid complicating the signature, especially if using the prefix middleware is optional. If it is absolutely required for normal operation of this API then it shouldn't be an option but rather an argument as you made it.

Reply:
The current plan is to pass this prefix for all requests (not just for UI ones) because this can be used for redirection etc in future. do we still want this as optional param?

Reply:
This param is now not required at all, guess that settles it :P